source/DEVS-Network/DEVSHostServiceResolver.class.st

```diff
@@ -0,0 +1,87 @@
+Class {
+	#name : 'DEVSHostServiceResolver',
+	#superclass : 'DEVSAtomicComponent',
+	#instVars : [
+		'hostname',
+		'address',
+		'ports',
+		'messages',
+		'state'
+	],
+	#category : 'DEVS-Network',
+	#package : 'DEVS-Network'
+}
+
+{ #category : 'as yet unclassified' }
+DEVSHostServiceResolver >> addServer: aServer [ 
+	ports 
+		at: aServer port 
+		put: (self addOutputPortNamed: aServer port )
+]
+
+{ #category : 'accessing' }
+DEVSHostServiceResolver >> address [
+	^ address
+]
+
+{ #category : 'accessing' }
+DEVSHostServiceResolver >> address: aString [ 
+	address := aString 
+]
+
+{ #category : 'as yet unclassified' }
+DEVSHostServiceResolver >> externalTransition [
+
+	| msg |
+	msg := self peekFrom: #in.
+	messages add: msg.
+	state := #processing 
+]
+
+{ #category : 'accessing' }
+DEVSHostServiceResolver >> hostname [
+	^ hostname
+]
+
+{ #category : 'initialization' }
+DEVSHostServiceResolver >> hostname: aString [ 
+	hostname := aString
+]
+
+{ #category : 'modeling' }
+DEVSHostServiceResolver >> initialize [
+	super initialize.
+	self addInputPortNamed: #in.
+	self addOutputPortNamed: #out.
+	ports := Dictionary new.
+	messages := OrderedCollection new
+]
+
+{ #category : 'as yet unclassified' }
+DEVSHostServiceResolver >> internalTransition [
+	messages ifEmpty: [  
+		state := #idle ]
+]
+
+{ #category : 'modeling' }
+DEVSHostServiceResolver >> outputFunction [
+	| msg |
+	(state = #processing) ifTrue: [ 
+		msg := messages removeFirst.
+		(msg destinationAddress = parent address)
```

Comment:
This check is for purpose of forwarding right? I just don't know when it may happen that destination address is not equal to parent address. Logic of router (its outputFunction) [pokes to](https://github.com/ApptiveGrid/PharoDEVS/pull/13/files#diff-93779361be1d43c38e74b89ed85d5d8426a80458b42ea732193e2767546c007eR56) correct address in current logic. So is it just for potential cases, when forwarding may happen (or will be supported)?

Reply:
This is needed because currently the server sends a message over the resolver out. The case is that only incoming messages have the same address not outgoing. Now that I finished my tests the coupling can be made differently and then this code can be removed